Release notes for v6.1.0 (#3296)
emqtt/emq-docs-cn: en_US/changes/changes-ee-v6.md

# EMQX Enterprise Version 6

## 6.1.0

*Release Date: 2025-12-30*

Make sure to check the breaking changes and known issues before upgrading to EMQX 6.1.0.

### Feature Highlights

> EMQX 6.1.0 introduces Message Streams, enhanced namespace capabilities, new data integrations, and centralized certificate management.

Reviewer: ```suggestion
EMQX 6.1.0 introduces MQTT Streams, enhanced namespace capabilities, new data integrations, and centralized certificate management.
```

Author: Does it means we also need to change the documents and Dashboard UI?

> **Message Streams**

Reviewer: ```suggestion
**MQTT Streams**
```

> EMQX Message Streams feature provide durable collections of messages identified by a topic filter, with explicit lifecycle management. Messages matching a stream's topic filter are automatically appended, enabling consumption with ordering guarantees and support for multiple consumers. Clients can subscribe to streams using the special topic format `$s/<timestamp>/topic/filter` to consume messages from a specific point in time.

Reviewer: ```suggestion
MQTT Streams feature provide durable collections of messages identified by a topic filter, with explicit lifecycle management. Messages matching a stream's topic filter are automatically appended, enabling consumption with ordering guarantees and support for multiple consumers. Clients can subscribe to streams using the special topic format `$s/<timestamp>/topic/filter` to consume messages from a specific point in time.
```

**Enhanced Namespace Capabilities**

- Configurations for namespace and isolation settings are now grouped together in the dashboard.
- Expanded namespace functionality with namespaced metrics, authentication, and authorization.
- Namespaced metrics are now available for messages, sessions, and data integration operations, exposed via Prometheus endpoints.
- Built-in authentication and authorization backends now support namespace-specific users and rules, enabling better multi-tenant isolation.
- Added automatic topic isolation using client namespaces as mountpoints.

**New Data Integrations**

- AWS Timestream for InfluxDB connector
- EMQX Tables connector
- InfluxDB API v3 support for InfluxDB and AWS Timestream connectors
- OAuth authentication for Kafka and Confluent Producer connectors
- Parquet file support for Azure Blob Storage and S3 Actions in Aggregated mode

**Certificate Management**

Added centralized certificate management via HTTP API, allowing certificates to be managed independently and referenced in SSL options for listeners and connectors.

### Enhancements

> #### Message Queue and Stream

Reviewer 2: ```suggestion
#### Message Queue and MQTT Stream
```

> - [#16326](https://github.com/emqx/emqx/pull/16326) Implemented Message Streams.

Reviewer: ```suggestion
- [#16326](https://github.com/emqx/emqx/pull/16326) Implemented MQTT Streams.
```

>   Message Streams are durable collections of messages identified by a topic filter.

Reviewer: ```suggestion
  MQTT Streams are durable collections of messages identified by a topic filter.
```

  They have an explicit lifecycle, and any published message that matches the Stream's topic filter is automatically appended to the stream.
  Streams allow consumption of messages with ordering guarantees and can be consumed multiple times.
  To consume messages from a stream, clients can subscribe to a special topic of the form
  `$s/<timestamp>/topic/filter`, where `topic/filter` refers to an existing stream. Subscribing with a timestamp allows consumption to begin at a specific point in time. The timestamp may be a Unix timestamp in microseconds or one of two special values: `earliest` or `latest`.

> - [#16454](https://github.com/emqx/emqx/pull/16454) For Message Queues and Streams, reconfigured garbage collection interval is now applied immediately. Previously, the new interval was applied only after the next garbage collection cycle.

Reviewer 2: ```suggestion
- [#16454](https://github.com/emqx/emqx/pull/16454) For Message Queues and MQTT Streams, reconfigured garbage collection interval is now applied immediately. Previously, the new interval was applied only after the next garbage collection cycle.
```
